benchmark_runner/common/clouds/shared/s3/s3_operations.py

```diff
@@ -153,10 +153,29 @@ def file_exist(self, bucket: str, key: str, file_name: str):
         except Exception:
             raise S3FileNotExist
 
+    @typeguard.typechecked
+    def folder_exist(self, bucket: str, key: str):
+        """
+        This method checks if folder/key exists in s3
+        :param bucket:'benchmark'
+        :param key:'framework/test'
+        :return:
+        """
+        try:
+            response = self.__s3_client.list_objects_v2(Bucket=bucket, Prefix=key)
+            if response.get('Contents'):
+                return True
+            return False
+        # Todo add custom error
+        except ClientError:
+            raise
```

Comment:
This is valid but confusing.  Is this specifically for the client error?  This would be more obvious (IMO):

```
except ClientError as exc:
    raise exc from null
```

But if you want to keep it this way I won't seriously object.

Reply:
This is the way to throw an exception to the upper level.